Source/WebCore/Modules/mediastream/libwebrtc/LibWebRTCStatsCollector.cpp

```diff
@@ -648,98 +632,75 @@ static inline void fillRTCAudioSourceStats(RTCStatsReport::AudioSourceStats& sta
-        } else if (rtcStats.type() == webrtc::RTCCodecStats::kType) {
-            RTCStatsReport::CodecStats stats;
-            fillRTCCodecStats(stats, static_cast<const webrtc::RTCCodecStats&>(rtcStats));
-            report.set<IDLDOMString, IDLDictionary<RTCStatsReport::CodecStats>>(stats.id, WTFMove(stats));
-        } else if (rtcStats.type() == webrtc::RTCTransportStats::kType) {
-            RTCStatsReport::TransportStats stats;
-            fillRTCTransportStats(stats, static_cast<const webrtc::RTCTransportStats&>(rtcStats));
-            report.set<IDLDOMString, IDLDictionary<RTCStatsReport::TransportStats>>(stats.id, WTFMove(stats));
-        } else if (rtcStats.type() == webrtc::RTCPeerConnectionStats::kType) {
-            RTCStatsReport::PeerConnectionStats stats;
-            fillRTCPeerConnectionStats(stats, static_cast<const webrtc::RTCPeerConnectionStats&>(rtcStats));
-            report.set<IDLDOMString, IDLDictionary<RTCStatsReport::PeerConnectionStats>>(stats.id, WTFMove(stats));
-        } else if (rtcStats.type() == webrtc::RTCAudioSourceStats::kType) {
-            RTCStatsReport::AudioSourceStats stats;
-            fillRTCAudioSourceStats(stats, static_cast<const webrtc::RTCAudioSourceStats&>(rtcStats));
-            report.set<IDLDOMString, IDLDictionary<RTCStatsReport::AudioSourceStats>>(stats.id, WTFMove(stats));
-        } else if (rtcStats.type() == webrtc::RTCVideoSourceStats::kType) {
-            RTCStatsReport::VideoSourceStats stats;
-            fillRTCVideoSourceStats(stats, static_cast<const webrtc::RTCVideoSourceStats&>(rtcStats));
-            report.set<IDLDOMString, IDLDictionary<RTCStatsReport::VideoSourceStats>>(stats.id, WTFMove(stats));
-        } else if (rtcStats.type() == webrtc::RTCRemoteInboundRtpStreamStats::kType) {
-            RTCStatsReport::RemoteInboundRtpStreamStats stats;
-            fillRemoteInboundRtpStreamStats(stats, static_cast<const webrtc::RTCRemoteInboundRtpStreamStats&>(rtcStats));
-            report.set<IDLDOMString, IDLDictionary<RTCStatsReport::RemoteInboundRtpStreamStats>>(stats.id, WTFMove(stats));
-        } else if (rtcStats.type() == webrtc::RTCRemoteOutboundRtpStreamStats::kType) {
-            RTCStatsReport::RemoteOutboundRtpStreamStats stats;
-            fillRemoteOutboundRtpStreamStats(stats, static_cast<const webrtc::RTCRemoteOutboundRtpStreamStats&>(rtcStats));
-            report.set<IDLDOMString, IDLDictionary<RTCStatsReport::RemoteOutboundRtpStreamStats>>(stats.id, WTFMove(stats));
-        } else if (rtcStats.type() == webrtc::RTCAudioPlayoutStats::kType) {
-            RTCStatsReport::AudioPlayoutStats stats;
-            fillRTCAudioPlayoutStats(stats, static_cast<const webrtc::RTCAudioPlayoutStats&>(rtcStats));
-            report.set<IDLDOMString, IDLDictionary<RTCStatsReport::AudioPlayoutStats>>(stats.id, WTFMove(stats));
-        }
+        if (rtcStats.type() == webrtc::RTCInboundRtpStreamStats::kType)
```

Comment:
This seems less than ideal

Reply:
Switch maybe?

Reply:
type() is a const char*, it should indeed be migrated to an enum in libwebrtc, in the meantime, we do not have the choice really.